src/erlfmt.erl

```diff
@@ -423,6 +429,16 @@ try_location(_, Node) when is_tuple(Node) -> erlfmt_scan:get_anno(location, Node
 try_location(_, [Node | _]) when is_tuple(Node) -> erlfmt_scan:get_anno(location, Node);
 try_location(_, _) -> 0.
 
+write_formatted(FileName, Formatted, check) ->
+    Original = read_file(FileName, fun(File) ->
+        erlfmt_scan:read_all(File)
+    end),
+    FormattedStr = unicode:characters_to_list(Formatted),
+    OriginalStr = unicode:characters_to_list(Original),
+    case OriginalStr =:= FormattedStr of
```

Comment:
With `file:read_file` we get a single binary back, we don't need to convert it, if we also convert formatted to a binary (this should also be slightly faster).
```suggestion
    FormattedStr = unicode:characters_to_binary(Formatted),
    case Original =:= FormattedStr of
```

Reply:
See the above comment.

Reply:
But I will try it out again, just to double check

Reply:
You were right, again :)
I will push a fix.